openquake/commonlib/logictree.py

```diff
@@ -797,27 +934,117 @@ def validate_uncertainty_value(self, node, branchset, value):
-                    rates = []
-                if _float_re.match(min_mag) and _float_re.match(bin_width) and\
-                        len(rates):
-                    return
+            min_mag, bin_width = (node.incrementalMFD["minMag"],
+                                  node.incrementalMFD["binWidth"])
+            rates = node.incrementalMFD.occurRates.text.strip()
+            try:
+                rates = valid.positivefloats(rates)
+            except ValueError:
+                rates = []
+            if _float_re.match(min_mag) and _float_re.match(bin_width) and\
+                    len(rates):
+                return
             raise ValidationError(
                 node, self.filename,
-                'expected mfd in the form min_mag,bin_width,rate_1 rate_2 ...'
+                "expected valid 'incrementalMFD' node"
             )
+        elif branchset.uncertainty_type == 'simpleFaultGeometryAbsolute':
+            self._validate_simple_fault_geometry(node.simpleFaultGeometry,
+                                                 _float_re)
+        elif branchset.uncertainty_type == 'complexFaultGeometryAbsolute':
+            self._validate_complex_fault_geometry(node.complexFaultGeometry,
+                                                  _float_re)
+        elif branchset.uncertainty_type ==\
+                'characteristicFaultGeometryAbsolute':
+            for geom_node in node.surface:
+                if "simpleFaultGeometry" in geom_node.tag:
+                    self._validate_simple_fault_geometry(geom_node, _float_re)
+                elif "complexFaultGeometry" in geom_node.tag:
+                    self._validate_complex_fault_geometry(geom_node, _float_re)
+                elif "planarSurface" in geom_node.tag:
+                    self._validate_planar_fault_geometry(geom_node, _float_re)
+                else:
+                    raise ValidationError(
+                        geom_node, self.filename,
+                        "Surface geometry type not recognised")
         else:
-            if not _float_re.match(value):
+            if not _float_re.match(node.text.strip()):
                 raise ValidationError(
                     node, self.filename,
                     'expected single float value'
                 )
 
+    def _validate_simple_fault_geometry(self, node, _float_re):
+        """
+        Validates a node representation of a simple fault geometry
+        """
+        usd, lsd, dip = (node.upperSeismoDepth.text.strip(),
+                         node.lowerSeismoDepth.text.strip(),
+                         node.dip.text.strip())
+        spacing = node["spacing"].strip()
+        try:
+            # Parse the geometry
+            coords = split_coords_2d(map(float,
+                                         node.LineString.posList.text.split()))
+            trace = geo.Line([geo.Point(*p) for p in coords])
+        except ValueError:
+            trace = []
```

Comment:
Same as before. The trace disappear while I would have expected a ValidationError to be raised. Am I missing something?